src/material.cpp

```diff
@@ -221,19 +227,41 @@ void pyne::Material::from_hdf5(std::string filename, std::string datapath, int r
   // Open the database
   hid_t db = H5Fopen(filename.c_str(), H5F_ACC_RDONLY, fapl);
 
-  bool datapath_exists = h5wrap::path_exists(db, datapath);
-  if (!datapath_exists)
-    throw h5wrap::PathNotFound(filename, datapath);
 
   // Clear current content
   comp.clear();
-
   // Load via various protocols
-  if (protocol == 0)
+  if (protocol == 0) {
+    bool datapath_exists = h5wrap::path_exists(db, datapath);
+    if (!datapath_exists)
+      throw h5wrap::PathNotFound(filename, datapath);
     _load_comp_protocol0(db, datapath, row);
-  else if (protocol == 1)
-    _load_comp_protocol1(db, datapath, row);
-  else
+  } else if (protocol == 1) {
+    
+    // Check /material type
+    herr_t status;
+    H5O_info_t object_info;
+    status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+    status = H5Oget_info_by_name(db, "/material" , &object_info, H5P_DEFAULT);
+    herr_t status2;
+    H5O_info_t object_info2;
+    status2 = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+    status2 = H5Oget_info_by_name(db, datapath.c_str(), &object_info, H5P_DEFAULT);
+    // Group "/material" does not exist
```

Comment:
Had to add a check for H5_DATASET for in the custom `datapath` to the one in the default `datapath` (`/material`)